src/sleep/sleep.c

```diff
@@ -1,6 +1,48 @@
+#include <ctype.h>
+#include <signal.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 
-int main(void) {
-  printf("Hello, World!\n");
-  return 0;
-}
\ No newline at end of file
+#define NAME "sleep (canoutils)"
+#define VERSION "1.0.0"
+#define AUTHOR "Eike Flath"
+
+#include "version_info.h"
+
+static unsigned int parse_int(char *arg) {
+  unsigned int n = 0;
+  for (int i = 0; arg[i]; i++) {
+    if (!isdigit(arg[i])) {
+      fprintf(stderr, "sleep: invalid time interval '%s'\n", arg);
+      exit(EXIT_FAILURE);
+    }
+    unsigned int new_n = n * 10 + (arg[i] - '0');
+    if (new_n < n) { // overflow
+      fprintf(stderr, "sleep: time interval '%s' is too big\n", arg);
+      exit(EXIT_FAILURE);
+    }
+    n = new_n;
+  }
+  return n;
+}
```

Comment:
Use `strtol` (C99, section `7.20.1.3`)

Reply:
Thanks for the review! I implemented the suggested change (see my latest commit)